### PR TITLE
[fix/response-transformer] concatenating value to response header

### PR DESCRIPTION
**Pull request by shashiranjan84**
Part of the fix for issue #393. New property `concat` added to support adding new value to existing response header. If header  does not exist, a new one will be added

### CODE REVIEW

**Comment by thibaultCha** [comment]
I have 2 suggestions:

#393 was mainly complaining that if the header already exists, then the value is overwritten by `add`. I think it would be necessary to fix this behavior first. `add` should *not* modify a header if it already exists.

This way:
- `add` only does so if the header does not already exists
- `concat` creates a header or adds a new value to it (eventually making it appear multiple times).

Since it is not obvious that `concat` also adds the header if it is not existing. I suggest renaming it to `append`.

So to recap:
- 1. make `add` not override an already existing header
- 2. rename `concat` to `append`

And finally: is this documented in the plugin's doc?

**Comment by thibaultCha** [comment]
up, since we will want this included in the next release.

**Comment by shashiranjan84** [comment]
What if user intentionally want to add new value to existing header?

**Comment by thibaultCha** [comment]
`replace`.

**Comment by shashiranjan84** [comment]
so 1st todo is invalid here as it's already happening.

**Comment by thibaultCha** [comment]
> What if user intentionally want to add new value to existing header?

I read that wrong I think, but this is the new setting: `concat` (or `append` which I prefer for the reasons described).

I just modify this plugin's behavior because:

> #393 was mainly complaining that if the header already exists, then the value is overwritten by add. I think it would be necessary to fix this behavior first. **`add` should not modify a header if it already exists.**

So, a new property is fine, but fixing the behavior of `add` should also be addressed. Basically we have isolated 3 use cases:

- Add a value to an existing header (so, create another header with the same name and a different value) (the main fix of this PR): `append` (or `concat` but like I said, I think `append` because this creates the header if it does not exist)
- Add a header **if it this header does not already exist**: `add`. This is a new change I am proposing, because the user complained that this setting was overwriting already existing headers.
- Unconditionally replace a header's value: `replace` then, since `add` only operates when the header does not exist.

**Comment by shashiranjan84** [comment]
> Unconditionally replace a header's value: `replace` then, since add only operates when the header does not exist.

what will `replace` do if header does not exist? 

How about `add` acts as append and `replace` used for replacing existing value and adding a new one if it does not exists.

**Comment by shashiranjan84** [comment]
PR can be reviewed for merge.

**Comment by thibaultCha** [comment]
This is opened against master?

**Comment by shashiranjan84** [comment]
it was a fix, so I opend PR against master. I will open a new PR against next.
